syncer/syncer.go

```diff
@@ -2833,9 +2835,12 @@ func (s *Syncer) handleQueryEventPessimistic(qec *queryEventContext) error {
 
 // input `sql` should be a single DDL, which came from parserpkg.SplitDDL
 // tableNames[0] is source (upstream) tableNames, tableNames[1] is target (downstream) tableNames.
-func (s *Syncer) trackDDL(usedSchema string, sql string, tableNames [][]*filter.Table, stmt ast.StmtNode, ec *eventContext) error {
-	srcTables, targetTables := tableNames[0], tableNames[1]
-	srcTable := srcTables[0]
+func (s *Syncer) trackDDL(usedSchema string, trackInfo *trackInfo, ec *eventContext) error {
```

Comment:
maybe use `qec` as trackDDL's param?

Reply:
I don't think so. `trackDDL` only track single `DDL`. In `qec`, there are `[]*trackInfo`.

Reply:
ok, got that, can we change `trackDDL` to `trackDDLs` ?

Reply:
We can't change `trackDDL` to `trackDDLs` simply. Because `handleQueryEventOptimistic` will do some other operations https://github.com/pingcap/dm/blob/master/syncer/optimist.go#L123-L134.